src/main/java/Parser.java

```diff
@@ -0,0 +1,161 @@
+
+        userTasks[userTaskIndex] = new ToDos(detail);
+
+        msg = "Gotcha. Do this while you're at it:\n"
+                + "\t\t" + userTasks[userTaskIndex].toString() + '\n'
+                + "\tNow you have " + (userTaskIndex + 1)
+                + " tasks in the list.";
+
+        userTaskIndex++;
+        return msg;
+    }
+
+    private static String parseDeadlineCommand(String command) {
+        String msg;
+        String detail = command.substring(COMMAND_DEADLINE.length()).trim();
+        String[] contentAndDate = detail.split(COMMAND_DEADLINE_SEPARATOR);
+
+        for (int i = 0; i < contentAndDate.length; i++) {
+            if (contentAndDate.length != EVENT_DEADLINE_ARGUMENT_COUNT
+                    || contentAndDate[i].equals(EMPTY_STRING)) {
+                msg = "Invalid format. Enter by this format:\n"
+                        + "\t\t\"deadline [description] /by [deadline]\"";
+                return msg;
+            }
+        }
+
+        for (int i = 0; i < contentAndDate.length; i++) {
+            contentAndDate[i] = contentAndDate[i].trim();
+        }
+
+        userTasks[userTaskIndex] = new Deadline(contentAndDate[0], contentAndDate[1]);
+
+        msg = "Gotcha. I beg you to do this:\n"
+                + "\t\t" + userTasks[userTaskIndex].toString() + '\n'
+                + "\tNow you have " + (userTaskIndex + 1)
+                + " tasks in the list.";
+
+        userTaskIndex++;
+        return msg;
+    }
+
+    private static String parseEventCommand(String command) {
+        String msg;
+        String detail = command.substring(COMMAND_EVENT.length()).trim();
+        String[] contentAndDate = detail.split(COMMAND_EVENT_SEPARATOR);
+
+        for (int i = 0; i < contentAndDate.length; i++) {
+            if (contentAndDate.length != EVENT_DEADLINE_ARGUMENT_COUNT
+                    || contentAndDate[i].equals(EMPTY_STRING)) {
+                msg = "Invalid format. Enter by this format:\n"
+                        + "\t\t\"event [description] /at [date]\"";
+                return msg;
+            }
+        }
+
+        for (int i = 0; i < contentAndDate.length; i++) {
+            contentAndDate[i] = contentAndDate[i].trim();
+        }
+
+        userTasks[userTaskIndex] = new Event(contentAndDate[0], contentAndDate[1]);
+
+        msg = "Gotcha. You wanna attend this:\n"
+                + "\t\t" + userTasks[userTaskIndex].toString() + '\n'
+                + "\tNow you have " + (userTaskIndex + 1)
+                + " tasks in the list.";
+
+        userTaskIndex++;
+        return msg;
+    }
+
+    private static String parseDoneCommand(String command) {
+        String msg;
+        String detail = command.substring(COMMAND_DONE.length()).trim();
+
+        int taskDoneNumber = Integer.parseInt(detail);
+
+        if (taskDoneNumber <= 0 || taskDoneNumber > userTaskIndex) {
+            msg = "Task number not exist!";
+            return msg;
+        }
```

Comment:
Good use of happy path to deal with unusual conditions of the task index to be marked